src/main/java/net/sf/rails/game/PublicCompany.java

```diff
@@ -35,7 +35,7 @@
  */
 public class PublicCompany extends RailsAbstractItem implements Company, RailsMoneyOwner, PortfolioOwner, Comparable<PublicCompany> {
 
-    private static final Logger log = LoggerFactory.getLogger(PublicCompany.class);
+    protected static final Logger log = LoggerFactory.getLogger(PublicCompany.class);
```

Comment:
`private`

Reply:
Hm why ? theres a PublicCompany_1844 needing access to that logger...

Reply:
Loggers should always be private and *only* available to the class declared in. If a subclass needs a logger, it should get it's own :)

The most obvious reason is that if you use a logger from a different class the detail included in the log will be incorrect-- it, for instance, would show the wrong "logger name" giving the indication the log line was generated a class different than it actually was, ie a log line like this would end up misleading:
```
05-18-2020:17:21:32.804 [AWT-EventQueue-0] DEBUG n.s.r.u.s.GameUIManager->processOnServer():612 -  - ==Result from server: true
```

As the creation of the logger is a one-time hit during class construction (not even instance construction) there is no real reason to not declare a logger for each class that needs to log.